server/src/server.js

```diff
@@ -1091,16 +1091,22 @@ app.get("/api/fxa-oauth/login/*", async function(req, res, next) {
   try {
     const stateBytes = await randomBytes(32);
     let state = stateBytes.toString("hex");
-    let stateId = null;
 
-    if (req.deviceId) {
+    const cookies = new Cookies(req, res, {keys: dbschema.getKeygrip("fxa-oauth")});
+    let stateId = cookies.get("fxaState");
+    if (stateId && (stateId.search(/^[a-zA-Z0-9_-]{1,255}$/) === -1)) {
```

Comment:
Arguably it's clearer to say `/^[a-zA-Z0-9_-]{1,255}$/.test(stateId)`, but that's just a minor nit.

Reply:
@punamdahiya I'm not sure this is the right regexp to verify a UUID is well-formed